homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,229 @@
+EVENT_ZHONG_HONG_HUB_START = 'zhong_hong_hub_start'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST):
+    cv.string,
+    vol.Optional(CONF_PORT, default=9999):
+    vol.Coerce(int),
+    vol.Optional(CONF_GATEWAY_ADDRRESS, default=1):
+    vol.Coerce(int),
+})
+
+
+async def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZhongHong HVAC platform."""
+    from zhong_hong_hvac.hub import ZhongHongGateway
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    gw_addr = config.get(CONF_GATEWAY_ADDRRESS)
+    hub = ZhongHongGateway(host, port, gw_addr)
+    try:
+        devices = [
+            ZhongHongClimate(hub, addr_out, addr_in)
+            for (addr_out, addr_in) in hub.discovery_ac()
+        ]
+
+    except Exception as exc:
+        _LOGGER.error("ZhongHong controller is not ready", exc_info=exc)
+        raise PlatformNotReady
+
+    device_num = len(devices)
+    _LOGGER.debug("We got %s zhong_hong climate devices", device_num)
+    device_ready_num = 0
+    semaphore_lock = asyncio.Semaphore()
+
+    async def device_count(event):
+        """Count ready device and fire hub start event."""
+        nonlocal device_ready_num
+        with (await semaphore_lock):
+            device_ready_num += 1
+            _LOGGER.debug("%s zhong_hong device setted up", device_ready_num)
+            if device_ready_num >= device_num:
+                hass.bus.fire(EVENT_ZHONG_HONG_HUB_START)
+
+    async_remove_listener = hass.bus.async_listen(EVENT_DEVICE_SETTED_UP,
+                                                  device_count)
+    # add devices after EVENT_DEVICE_SETTED_UP event is listend
+    add_devices(devices)
+
+    def startup(event):
+        """Start hub socket after all climate entity is setted up."""
+        _LOGGER.debug("zhong_hong hub start listen event")
+        async_remove_listener()
+        hub.start_listen()
+        hub.query_all_status()
+
+    hass.bus.async_listen_once(EVENT_ZHONG_HONG_HUB_START, startup)
+
+    def stop_listen(event):
+        """Stop ZhongHongHub socket."""
+        hub.stop_listen()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_listen)
+
+
+class ZhongHongClimate(ClimateDevice):
+    """Representation of a ZhongHong controller support HVAC."""
+
+    def __init__(self, hub, addr_out, addr_in):
+        """Set up the ZhongHong climate devices."""
+        from zhong_hong_hvac.hvac import HVAC
+        self._device = HVAC(hub, addr_out, addr_in)
+        self._hub = hub
+        self._current_operation = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._current_fan_mode = None
+        self._is_on = None
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        self._device.register_update_callback(self._after_update)
+        self.hass.bus.fire(EVENT_DEVICE_SETTED_UP)
```

Comment:
Send the message to the target via dispatcher here to let he target know that this device has been added.